docs/building-blocks/extending-templates/serial-copy-loop.md

```diff
@@ -0,0 +1,150 @@
+---
+title: Extending serial copy functionality in a resource manager template
+description: Describes how to extend the functionality of a serial copy loop in an Azure Resource Manager template
+author: petertay
+ms.date: 05/03/2017
+
+---
+
+# Extending serial copy functionality in a resource manager template
+
+You can deploy multiple instances of a single resource type in an [Azure Resource Manager][azure-resource-manager-overview] template using a [copy loop][create-multiple-instances]. To do this, you specify the copy loop within the resource. Then, resource manager processes the resource as many times as specified in the `count` property. The resource's properties are dynamically set by assigning them to a parameter or variable array that is indexed using the [`copyIndex()`][azure-resource-manager-functions-numeric] function. For example:
+
+```json
+  ...
+  "parameters" : {
+    "domain" : {
+      "type": "array",
+      "defaultValue": [ "contoso.com","fabrikam.com"]
+    }
+  ...
+  "resources": {
+    "name": "[concat('domain',parameters('domain')[copyIndex()])]",
+    "copy": {
+      "name": "domainCopy",
+      "count": 2,
+      "mode": "serial"
+    }
+    "properties": {
+      "domain": "[parameters('domain')[copyIndex()]]"
+    }
+  }
+}
+```
+Note that resource names must be unique within a single deployment, so the resource's name is also dynamically set using the `copyIndex()` function.
+
+Also note that setting the `mode` property to `serial` specifies that the copy loop should be sequential. That is, the default behavior of resource manager is to deploy all resources in parallel - including those in a copy loop - but in a `serial` copy loop, resource manager waits until the last resource in the loop is successfully deployed before moving on to the next.
+
+This functionality is very useful in scenarios in which child resources have to be applied to a parent resource one at a time, such as when a network security rule is applied to a network security group (NSG). However, there are instances in which it's useful to refer to the output of a previous resource within a loop. For example, you may wish to output an array of properties to validate that they were applied correctly.
```

Comment:
I would explain why this is useful. If you are adding child resources to a parent resource in parallel, each child locks the resource for update, so there is a concurrency issue.